flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveTableCompactSinkTest.java

```diff
@@ -0,0 +1,195 @@
+
+    private static HiveCatalog catalog;
+
+    private Configuration tableConf;
+
+    @BeforeAll
+    static void before() {
+        catalog = HiveTestUtils.createHiveCatalog();
+        catalog.open();
+    }
+
+    @AfterAll
+    static void after() {
+        catalog.close();
+    }
+
+    @BeforeEach
+    void resetTableConf() {
+        tableConf = new Configuration();
+        tableConf.set(FactoryUtil.CONNECTOR, SqlCreateHiveTable.IDENTIFIER);
+        tableConf.set(FileSystemConnectorOptions.AUTO_COMPACTION, true);
+    }
+
+    /** If only sink parallelism is set, compact operator should follow this setting. */
+    @Test
+    void testOnlySetSinkParallelism() throws Exception {
+        final int sinkParallelism = 4;
+
+        tableConf.set(FileSystemConnectorOptions.SINK_PARALLELISM, sinkParallelism);
+
+        assertSinkAndCompactOperatorParallelism(
+                tableConf, true, true, sinkParallelism, sinkParallelism);
+    }
+
+    @Test
+    void testOnlySetCompactParallelism() throws Exception {
+        final int compactParallelism = 4;
+
+        tableConf.set(FileSystemConnectorOptions.COMPACTION_PARALLELISM, compactParallelism);
+
+        assertSinkAndCompactOperatorParallelism(tableConf, false, true, -1, compactParallelism);
+    }
+
+    @Test
+    void testSetBothSinkAndCompactParallelism() throws Exception {
+        final int sinkParallelism = 8;
+        final int compactParallelism = 4;
+
+        tableConf.set(FileSystemConnectorOptions.SINK_PARALLELISM, sinkParallelism);
+        tableConf.set(FileSystemConnectorOptions.COMPACTION_PARALLELISM, compactParallelism);
+
+        assertSinkAndCompactOperatorParallelism(
+                tableConf, true, true, sinkParallelism, compactParallelism);
+    }
+
+    @Test
+    void testSinkAndCompactAllNotSetParallelism() throws Exception {
+        assertSinkAndCompactOperatorParallelism(tableConf, false, false, -1, -1);
+    }
+
+    private void assertSinkAndCompactOperatorParallelism(
+            Configuration tableConf,
+            boolean isSinkParallelismConfigured,
+            boolean isCompactParallelismConfigured,
+            int expectedSinkParallelism,
+            int expectedCompactParallelism)
+            throws Exception {
+        StreamGraph streamGraph = createAndGetStreamGraphContainsSink(tableConf.toMap());
+        assertThat(streamGraph.getStreamNodes()).hasSize(6);
+        StreamNode hiveSinkNode =
+                getFirstStreamNodeByName(streamGraph, HiveTableSink.BATCH_COMPACT_WRITER_OP_NAME);
+        StreamNode compactNode = getFirstStreamNodeByName(streamGraph, BatchSink.COMPACT_OP_NAME);
+        assertThat(hiveSinkNode.isParallelismConfigured()).isEqualTo(isSinkParallelismConfigured);
+        assertThat(compactNode.isParallelismConfigured()).isEqualTo(isCompactParallelismConfigured);
+        if (isSinkParallelismConfigured) {
+            assertThat(hiveSinkNode.getParallelism()).isEqualTo(expectedSinkParallelism);
```

Comment:
Can we consider introducing a field `private static final int DEFAULT_PARALLELISM`? When isXXXParallelismConfigured is false, the parallelism of the operator is equal to DEFAULT_PARALLELISM, which is also in line with the idea of table planner operator parallelism infection. If so, you may need to use fromSource because it is a parallel source.
What do you think?

Reply:
I have reworked this test logical to not dependent on `StreamGraph`, so I think control the default parallelism is a little difficult. BTW, I use a special static final field to indicate the parallelism check should be ignored.